main.go

```diff
@@ -280,12 +281,21 @@ func getBotChannels(api *slack.Client) ([]Channel, error) {
 	return results, nil
 }
 
-func isUpdateInChannelHistory(update StatusPageIncidentUpdate, api *slack.Client, channel Channel) (bool, error) {
+func isUpdateInChannelHistory(update StatusPageIncidentUpdate, api *slack.Client, channel Channel, retry int) (bool, error) {
 	historyResp, err := api.GetConversationHistory(&slack.GetConversationHistoryParameters{
 		ChannelID: channel.ID,
 	})
 	if err != nil {
 		log.Printf("Error getting conversation history: %v", err)
+		// If the error is because we hit a rate limit, sleep for the given time and try again
+		switch err.(type) {
+		case *slack.RateLimitedError:
+			if retry < retryMax {
+				time.Sleep(err.(*slack.RateLimitedError).RetryAfter)
+				return isUpdateInChannelHistory(update, api, channel, retry+1)
```

Comment:
I haven't looked deep into the code, but just thought I would ask as I have never implemented anything recursive in go - with channels. Is using recursion and channels ok? (I have always used things like wait groups)?

Reply:
Hmm good point, I'll check that. FWIW, the `channel` here isn't a go channel construct, its the slack channel ID so this isn't explicitly using a channel.

But I'll confirm there are channels in use or if it could be an issue in the recursion

Reply:
The only use of channels in this repo is https://github.com/buildkite/statusbot/blob/1b154a88653f6a5d959ecb14c6b7e9cf9b8c66d9/main.go#L74 which is for realtime messages and doesn't interact with this part of the code (which receives webhooks from StatusPage and posts messages to slack). The realtime side is what listens for joining new channels, etc to add them to the list to post to